### PR TITLE
chore(flowconfig, Components): Fixed Immutable related Flow errors

### PR DESCRIPTION
**Pull request by peggyrayzis**
@rgbkrk Hey Kyle! Looks like I spoke too soon on your last PR - putting the wildcard after the dist/\* folder caused Flow to throw a module not found error for Immutable. I did some digging and found that Immutable's Flow types are located in a dist/ folder, so I limited the scope of flowconfig to just our dist folder.

Also cleaned up the ImmutableMap typing to include 2 args - if the map values are more specific than any, lmk and I can adjust those definitions. I haven't used Immutable much, let alone w Flow, so this is a learning process for me :)

### CODE REVIEW

**Comment by rgbkrk** [comment]
Ohhhhh OK, thanks.

**Comment by rgbkrk** [comment]
Do you have any idea why Rx is not showing up in my `flow-typed` folder:

```
$ head flow-typed/npm/rxjs_vx.x.x.js
// flow-typed signature: 92666ab12556f8797740e7ed3562dae4
// flow-typed version: <<STUB>>/rxjs_v^5.0.0-beta.12/flow_v0.33.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   'rxjs'
 *
 * Fill this stub out by replacing all the `any` types.
 *
```

I see that there are Rx definitions inside of https://github.com/flowtype/flow-typed/tree/master/definitions/npm/rxjs_v5.0.x. Is it related to how we pin it?

File: src/notebook/components/cell/cell.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ type Props = {
   running: boolean,
   theme: string,
   pagers: ImmutableList<any>,
-  transforms: ImmutableMap<any>,
+  transforms: ImmutableMap<string, any>,
```
Comment (rgbkrk) [inline src/notebook/components/cell/cell.js:26]: These all map to React Components, which can be the big ol' classes or the stateless function.

File: src/notebook/components/cell/code-cell.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
+// @flow
 import React from 'react';
```
Comment (rgbkrk) [inline src/notebook/components/cell/code-cell.js:2]: Also, uh-oh, we're totally going to conflict in https://github.com/nteract/nteract/pull/928.

**Comment by peggyrayzis** [comment]
No idea why Rx isn't working for you, I'll have to look into it. I did notice that the Rx definitions are only tested against flow_v0.29.0-v0.32.x whereas we're running v0.33. Not sure how big of a difference that makes, but my coworker & I have seen some major discrepancies btwn versions before.

File: src/notebook/components/cell/code-cell.js (hunk 2 above)
Comment (rgbkrk) [reply src/notebook/components/cell/code-cell.js:2]: annnd maybe not, looks like we'll be good to go after CI passes

Comment (peggyrayzis) [reply src/notebook/components/cell/code-cell.js:2]: No biggie, do you want to merge first?

Comment (rgbkrk) [reply src/notebook/components/cell/code-cell.js:2]: @captainsafia already merged and it looks like we're ready to go here!

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Now that we're in this state, we can probably do https://github.com/nteract/nteract/issues/896

**Pull request merged**